lib/webtask.js

```diff
@@ -665,6 +680,39 @@ exports.ensure_authenticated = function(ctx) {
                         else if (req.claims.pctx && req.claims.pctx['webtask_' + claim])
                             req.claims[claim] = req.claims.pctx['webtask_' + claim];
                     });
+
+                    if(ectx && ectx.WEBTASK_JWT_SECRET && ectx.WEBTASK_JWT_AUD) {
+                        if(!req.headers['authorization'] && !req.query.key) {
+                            msg = { code: 401, message: 'must provide authorization header or key query', req_id: req.req_id };
+                            ctx.logger.error({
+                                req_id: req.req_id,
+                                path: req.originalUrl,
+                                method: req.method,
+                                keyset: jwt.header.kid,
+                                payload: jwt.payload,
+                                credential: credential
+                            }, msg.message);
+                            return callback(msg);
+                        }
+
+                        try {
+                            jwt.payload.user = validate_auth_token(ectx.WEBTASK_JWT_SECRET, ectx.WEBTASK_JWT_AUD);
+
+                            delete ectx.WEBTASK_JWT_SECRET;
+                            delete ectx.WEBTASK_JWT_AUD;
```

Comment:
We should probably delete those before calling validate_auth_token in case it throws.